src/trading/buyer.py

```
@@ -216,3 +221,135 @@ async def _send_buy_transaction(
+        try:
+            client = await self.client.get_client()
+
+            tx_response = await client.get_transaction(
+                tx_signature, 
+                encoding="jsonParsed",
+                commitment="confirmed",
+                max_supported_transaction_version=0
+            )
+            
+            if not tx_response.value or not tx_response.value.transaction:
+                raise ValueError("Transaction not found")
```

Ruff (0.11.9)

Abstract `raise` to an inner function

(TRY301)